framework/sandstone.cpp

```diff
@@ -2170,8 +2173,8 @@ TestResult run_one_test(int *tc, const struct test *test, SandstoneApplication::
             if (sApp->total_retest_count > 0)
                 --sApp->total_retest_count;
             sApp->current_iteration_count = -iterations;
-            sApp->current_test_endtime = calculate_wallclock_deadline(sApp->current_test_duration);
             init_internal(test);
+            sApp->current_test_endtime = calculate_wallclock_deadline(sApp->current_test_duration, &sApp->current_test_starttime);
```

Comment:
Why are you passing the starttime here, when the previous code didn't?

Reply:
This allows to set both from exactly same point of time (t.i. end_time := (start_time := now()) + duration). Having it split might provide some unintentional delays.